### PR TITLE
Fix measurement calculation following Vanadium fix

### PR DESCRIPTION
**Pull request by deeglaze**
Vanadium did not respect the TDVF Metadata attribute field for whether a region should be measured or not. By properly accounting for it in the hypervisor, the RAM banks or not relevant to MRTD.

The CFV and TDHOB will still be relevant to RTMRs.

### CODE REVIEW

File: cmd/endorse.go [hunk 1]
```diff
@@ -77,8 +77,8 @@ func (f *endorseCommand) AddFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().AddGoFlag(
 		amdProductVar(&ec.SevSnp.Product, "snp_product", sgpb.SevProduct_SEV_PRODUCT_MILAN,
 			"SEV-SNP product line. One of Milan, Genoa"))
-	cmd.PersistentFlags().BoolVar(&ec.Tdx.IncludeEarlyAccept, "tdx_include_early_accept", false, "If true, adds MRTD values for all supported machine shapes with all memory configured for acceptance in the firmware.")
-	cmd.PersistentFlags().StringSliceVar(&ec.Tdx.MachineShapes, "tdx_machine_shapes", nil, "GCE machine shapes whose measurement-relevant configuration should be enumerated for endorsement.")
+	cmd.PersistentFlags().BoolVar(&ec.Tdx.IncludeEarlyAccept, "tdx_include_early_accept", false, "DEPRECATED: If true, adds MRTD values for all supported machine shapes with all memory configured for acceptance in the firmware.")
+	cmd.PersistentFlags().StringSliceVar(&ec.Tdx.MachineShapes, "tdx_machine_shapes", nil, "DEPRECATED: GCE machine shapes whose measurement-relevant configuration should be enumerated for endorsement.")
```
Comment (AdamCDunlap) [inline cmd/endorse.go:81]: The "DEPRECATED" message is pretty confusing. What should a user do if they see that the flag is deprecated? Just not use it? Use a different flag?

File: proto/endorsement/endorsement.pb.go [hunk 2]
```diff
@@ -21,9 +21,9 @@
 package endorsement
 
 import (
-	timestamp "github.com/golang/protobuf/ptypes/timestamp"
 	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
 	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
```
Comment (AdamCDunlap) [inline proto/endorsement/endorsement.pb.go:26]: Huh why is this change bundled in here?

File: proto/endorsement.proto [hunk 3]
```diff
@@ -73,11 +73,17 @@ message VMTdx {
     // The amount of RAM in GiB provided to the VM. This is relevant to the
     // construction of the measured TDHOB page that includes memory regions'
     // resource attributes.
+    //
+    // To be deprecated: The TDHOB is no longer measured in newer Google
```
Comment (AdamCDunlap) [inline proto/endorsement.proto:77]: When will it be deprecated? When the buggy vanadium is no longer in production?

File: ovmf/memory.go [hunk 4]
```diff
@@ -77,4 +77,6 @@ type GuestPhysicalRegion struct {
 type MaterialGuestPhysicalRegion struct {
 	GPR        GuestPhysicalRegion
 	HostBuffer []byte
+	// Attributes is a TDVF-only field to track whether the buffer is measured.
```
Comment (AdamCDunlap) [inline ovmf/memory.go:80]: This comment suggests that it's a boolean that should be named IsBufferMeasured. Can you clarify how it tracks whether the buffer is measured?

File: cmd/endorse.go
Comment (AdamCDunlap) [inline cmd/endorse.go:1]: The commit message is missing some words "By properly accounting for it in the hypervisor, the RAM banks or not relevant to MRTD."

I'm confused about what exactly this change is doing without reading the code. Can you update the commit message so it talks about how it detects if it's on the buggy version, why some options need to be removed, etc?

File: ovmf/memory.go (hunk 4 above)
Comment (deeglaze) [reply ovmf/memory.go]: Updated.

File: cmd/endorse.go (hunk 1 above)
Comment (deeglaze) [reply cmd/endorse.go]: Updated the text for the age of VMM needed for its relevance.

File: cmd/endorse.go
Comment (deeglaze) [reply cmd/endorse.go:1]: It cannot detect the buggy version. Added text to the flag that it's for Google VMM releases older than the one that's receiving the fix I'm trying to submit.

File: proto/endorsement.proto (hunk 3 above)
Comment (deeglaze) [reply proto/endorsement.proto:77]: Correct. I can remove it when the buggy VMM is irrelevant to account for.

File: proto/endorsement/endorsement.pb.go (hunk 2 above)
Comment (deeglaze) [reply proto/endorsement/endorsement.pb.go]: Moved protoc update to a different PR

File: ovmf/memory.go [hunk 5]
```diff
@@ -77,4 +77,6 @@ type GuestPhysicalRegion struct {
 type MaterialGuestPhysicalRegion struct {
 	GPR        GuestPhysicalRegion
 	HostBuffer []byte
+	// TDVFAttributes is a TDVF-only field for a bitset of directives tho the VMM.
```
Comment (AdamCDunlap) [inline ovmf/memory.go:80]: typo: "tho"

**Review by AdamCDunlap — APPROVED** [review]

**Pull request merged**